sympy/physics/continuum_mechanics/beam.py

```diff
@@ -1521,6 +1524,137 @@ def plot_loading_results(self, subs=None):
         return PlotGrid(4, 1, ax1, ax2, ax3, ax4)
 
 
+    @doctest_depends_on(modules=('numpy',))
+    def draw(self, pictorial=False):
+        """Returns a plot object representing the beam diagram of the beam.
+
+        Parameters
+        ==========
+
+        pictorial: Boolean (default=False)
+            Setting ``pictorial=True`` would simply create a pictorial (scaled) view
+            of the beam diagram not according to the scale.
+            Although setting ``pictorial=False`` would create a beam diagram with
+            the exact dimensions on the plot
+
+        Examples
+        ========
+
+        .. plot::
+            :context: close-figs
+            :format: doctest
+            :include-source: True
+
+            >>> from sympy.physics.continuum_mechanics.beam import Beam
+            >>> from sympy import symbols
+            >>> R1, R2 = symbols('R1, R2')
+            >>> E, I = symbols('E, I')
+            >>> b = Beam(50, 20, 30)
+            >>> b.apply_load(10, 2, -1)
+            >>> b.apply_load(R1, 10, -1)
+            >>> b.apply_load(R2, 30, -1)
+            >>> b.apply_load(90, 5, 0, 23)
+            >>> b.apply_load(10, 30, 1, 50)
+            >>> b.apply_support(50, "pin")
+            >>> b.apply_support(0, "fixed")
+            >>> b.apply_support(20, "roller")
+            >>> b.draw(pictorial=True)
+            Plot object containing:
+            [0]: cartesian line: 25*SingularityFunction(x, 5, 0)
+            - 25*SingularityFunction(x, 23, 0) + SingularityFunction(x, 30, 1)
+            - 20*SingularityFunction(x, 50, 0) - SingularityFunction(x, 50, 1)
+            + 5 for x over (0.0, 50.0)
+        """
+        if not numpy:
+            raise ImportError("To use this function numpy module is required")
+
+        x = self.variable
+        length = self.length
+        height = length/10
+
+        rectangles = []
+        rectangles.append({'xy':(0, 0), 'width':length, 'height': height, 'facecolor':"brown"})
+        annotations, markers, load_eq, fill = self._draw_load(pictorial)
+        support_markers, support_rectangles = self._draw_supports()
+
+        rectangles += support_rectangles
+        markers += support_markers
+
+        sing_plot = plot(height + load_eq, (x, 0, length),
+         xlim=(-2, length + 2), ylim=(-length, length), annotations=annotations,
```

Comment:
Would it make sense to use relative spacing instead of absolute? Like `(-0.05*length, 1.05*length)`?

Reply:
I tried `xlim=(-0.05*length, 1.05*length)`, it more or less gives similar output.

The main aim of having `xlim` is to include the rectangle (denoting the fixed support) in the plot.
Since the width of this rectangle is every time `length/10` (see line 1651), I think then we can rather go for `xlim=(-length/10, length + length/10)`.
I guess then even this would now be relative. 
Should we go for this then? @oscargus